github/secret_scanning.go

```diff
@@ -89,6 +89,11 @@ type SecretScanningAlertUpdateOptions struct {
 	// Required when the state is "resolved" and represents the reason for resolving the alert.
 	// Can be one of: "false_positive", "wont_fix", "revoked", or "used_in_tests".
 	Resolution *string `json:"resolution,omitempty"`
+
+	// An optional comment when closing an alert.
+	// Cannot be updated or deleted.
+	// Must be null when changing state to open.
```

Comment:
I copied the comment from the documentation. 
https://docs.github.com/en/rest/secret-scanning/secret-scanning?apiVersion=2022-11-28#update-a-secret-scanning-alert

However, I have now confirmed that comments can be added when changing to Open, 
so this comment may not be necessary.

Reply:
For me, it is better to be able to specify comments when changing to close state or open state.

When changing to open state, I think the comment should also be modified to match the actual behavior that can be specified as optional.

Therefore, how about removing the comment “Cannot be updated or deleted. Must be null when changing state to open.”

Reply:
Sounds good to me.